### DISCUSSION TITLE
state of lithiation and initial concentration setup

### DISCUSSION THREAD

**Discussion by jennya2a** [pybamm-team/PyBaMM#3902] [Q&A]
may I know how does pybamm calculate the relationship between state of lithiation in electrodes and the initial concentration in electrodes, is it similar to equations below and can you advise me where to find it in code?
cs_init_neg = csmax_neg * anode_min + csmax_neg * (anode_max - anode_min) * soc0
cs_init_pos = cathode_max * csmax_pos - csmax_pos * (cathode_max - cathode_min) * soc0

I also got a simulation issues related to it.
1. I optimized the max and min state of lithiation (anode_max, anode_min, cathode_max, cathode_min) in electrodes based on electrode ocp data and cell ocv-soc data.
2. then i feed them into model, and calculate initial concentration in electrodes using equations above, and update them in model parameters. soc0 is based on the initial voltage in test data and ocv-soc data.
3. then run the model, the simulated initial voltage is much higher than the initial voltage in test data. anyone could help me figure out the possible reasons? Thanks.

**Comment by valentinsulzer** [comment]
"Concentration" = "State of lithiation" * "Maximum concentration". For initialization see https://docs.pybamm.org/en/latest/source/examples/notebooks/models/electrode-state-of-health.html. Your approach sounds like it should work. For debugging try plotting things like "Negative electrode stoichiometry" and "Positive electrode stoichiometry" and see how they align with your calculations of min and max values
